### PR TITLE
Info about statistics cron job removal

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Drop WIP notice and add a note about automation of trends collection.
- [x] Depends on: https://github.com/theforeman/foreman_statistics/pull/19
- [ ] Packaging https://github.com/theforeman/foreman-packaging/pull/5625, https://github.com/theforeman/foreman-packaging/pull/5626

### CODE REVIEW

File: plugins/foreman_statistics/1.0/index.md [hunk 1]
```diff
@@ -6,17 +6,18 @@ version: 1.0
...
 # 1. {{ page.title }}
 
 This plugin enables Foreman to show a nice statistics dashboard and create Trend charts.
 
+## 1.1 Release notes
+
+### Drop the cron job
+
+Foreman statistics 1.0 onwards automates data collection and no manual steps are required.
+If you've set a cron job to execut `foreman-rake trends:counter` manually in the past, please remove it.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]:
```suggestion
If you've set a cron job to execute `foreman-rake trends:counter` manually in the past, please remove it.
```

File: plugins/foreman_statistics/1.0/index.md [hunk 2]
```diff
@@ -6,17 +6,18 @@ version: 1.0
...
-
 # 1. {{ page.title }}
 
 This plugin enables Foreman to show a nice statistics dashboard and create Trend charts.
 
+## 1.1 Release notes
+
+### Drop the cron job
+
+Foreman statistics 1.0 onwards automates data collection and no manual steps are required.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]: I guess it's using dynflow, but might be good to explicitly mention that.

File: plugins/foreman_statistics/1.0/index.md [hunk 3]
```diff
@@ -52,14 +53,8 @@ DEB:
...
-
-There are two pieces to the Trends area, the Trends to track and their corresponding counters. To define trend counters, use the "Add Trend Counter" button on the '/trends' page. Optionally set the "Name" field to over-ride odd puppet fact names to be more readable. Once created you can optionally 'Edit' the Trend to change the display names of the underlying values.
-
-Next, to start collecting trend data, set a cron job to execute 'foreman-rake trends:counter'. Each time the rake task executes it will create 1 tick on the graphs, so you can fine tune the granularity with your cron job. We recommend using the same as your puppet run interval (30 minutes). Here's an example to run once per hour:
+Trends in Foreman allow you to track changes in your infrastructure over time. It allows you to track both Foreman related information and any facts. The Trend pages give a graph of how the number of hosts with that value have changed over time, and the current hosts list.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]: I've read it's good practice to keep it impersonal
```suggestion
Trends in Foreman allow tracking changes in your infrastructure over time. It allows tracking both Foreman related information and any facts. The Trend pages give a graph of how the number of hosts have changed over time, and the current hosts list.
```

File: plugins/foreman_statistics/1.0/index.md [hunk 4]
```diff
@@ -52,14 +53,8 @@ DEB:
...
 
-{% highlight sh %}
-0 * * * * /usr/sbin/foreman-rake trends:counter
-{% endhighlight %}
+There are two pieces to the Trends area, the Trends to track and their corresponding counters. To define trend counters, use the "Add Trend Counter" button on the '/foreman_statistics/trends' page. Optionally set the "Name" field to over-ride odd fact names to be more readable. Once created you can optionally 'Edit' the Trend to change the display names of the underlying values.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]: Usually we don't give paths of URLs. Perhaps it should refer to it as a page title, like Trends Overview page?

Comment (ezr-ondrej) [reply plugins/foreman_statistics/1.0/index.md]: What about menu reference `Monitor > Trends`?

File: plugins/foreman_statistics/1.0/index.md (hunk 3 above)
Comment (ezr-ondrej) [reply plugins/foreman_statistics/1.0/index.md]: :+1: sounds better, thanks!

**Review by tbrisker — APPROVED** [review]
thanks @ezr-ondrej !

**Comment by tbrisker** [comment]
any other comments @ekohl ?

**Comment by ezr-ondrej** [comment]
The packaging is still not in, so it's technically true about the plugin 1.0, but there is no such version in our repositories :(

File: plugins/foreman_statistics/1.0/index.md [hunk 5]
```diff
@@ -6,17 +6,18 @@ version: 1.0
...
-
 # 1. {{ page.title }}
 
 This plugin enables Foreman to show a nice statistics dashboard and create Trend charts.
 
+## 1.1 Release notes
+
+### Drop the cron job
+
+Foreman statistics 1.0 onwards automates data collection by facilitating a dynflow processing engine and no manual steps are required.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]: I'm wondering if we need to mention the version. Versions prior to 1.0 weren't really shipped to users.
```suggestion
Foreman statistics moves trend calculation from a cronjob to a Dynflow job and no manual steps are required.
```

File: plugins/foreman_statistics/1.0/index.md [hunk 6]
```diff
@@ -6,17 +6,18 @@ version: 1.0
...
+If you've set a cron job to execute `foreman-rake trends:counter` manually in the past, please remove it.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]: Previously it was shipped in foreman itself. If users modified the cronjob, a yum update will not update it either. So even if they didn't modify the trends counter explicitly, they still need to look. So I'd phrase it a bit more open.
```suggestion
Users upgrading from older versions should ensure to remove any cronjobs that call `foreman-rake trends:counter`.
```

**Review by ekohl — COMMENTED** [review]
Not sure about the exact phrasing so feel free to come up with something different.

File: plugins/foreman_statistics/1.0/index.md [hunk 7]
```diff
@@ -6,17 +6,18 @@ version: 1.0
...
 # 1. {{ page.title }}
 
 This plugin enables Foreman to show a nice statistics dashboard and create Trend charts.
 
+## 1.1 Release notes
+
+### Drop the cron job
+
+Foreman statistics moves trend calculation from a cronjob to a Dynflow job and no manual steps are required.
+Users upgrading from older versions should ensure to remove any cronjobs that call `foreman-rake trends:counter`.
```
Comment (ekohl) [inline plugins/foreman_statistics/1.0/index.md]: Older versions may imply older versions of the plugin. May be call out that it was in core prior to Foreman 2.3?

Comment (ezr-ondrej) [reply plugins/foreman_statistics/1.0/index.md]: reworded to `from Foreman 2.2 and older`

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
@tbrisker mind giving it a last read and merge it if you think it's ok?

File: _includes/manuals/nightly/4.1.3_trends.md [hunk 8]
```diff
@@ -1,3 +1,5 @@
 
 Trends and statistics are moved to the separate plugin.
 See [manual](/plugins/foreman_statistics) for the reference.
+
+There is a rake task `foreman-rake purge:trends` for users who are not planning to use trends and statistics anymore and wish to clean up database.
```
Comment (tbrisker) [inline _includes/manuals/nightly/4.1.3_trends.md:5]: This should also be added in 2.2 manual (we shipped the task in rc3), perhaps also mention it in the 2.2 upgrade warning section mentioning the extraction.

File: plugins/foreman_statistics/1.0/index.md [hunk 9]
```diff
@@ -6,17 +6,18 @@ version: 1.0
...
 # 1. {{ page.title }}
 
 This plugin enables Foreman to show a nice statistics dashboard and create Trend charts.
 
+## 1.1 Release notes
+
+### Drop the cron job
+
+Foreman statistics moves trend calculation from a cronjob to a Dynflow job and no manual steps are required.
+Users upgrading from Foreman 2.2 and older versions should ensure to remove any cronjobs that call `foreman-rake trends:counter`.
```
Comment (tbrisker) [inline plugins/foreman_statistics/1.0/index.md:18]: This should also be mentioned in the 2.2 upgrade warning - if they are removing the plugin they should make sure to clean up the cronjob

**Review by tbrisker — CHANGES_REQUESTED** [review]
just a couple of things we might want to pull to the 2.2 manual as well.

File: plugins/foreman_statistics/1.0/index.md (hunk 9 above)
Comment (ezr-ondrej) [reply plugins/foreman_statistics/1.0/index.md:18]: done

File: _includes/manuals/nightly/4.1.3_trends.md (hunk 8 above)
Comment (ezr-ondrej) [reply _includes/manuals/nightly/4.1.3_trends.md:5]: Done

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !

**Pull request merged**